codalab/apps/newsletter/views.py

```diff
@@ -0,0 +1,110 @@
+import hashlib
+import json
+
+import requests
+from django.contrib import messages
+from django.contrib.sites.models import Site
+from django.shortcuts import render
+from django.core.mail import EmailMultiAlternatives
+from django.template import Context
+from django.template.loader import get_template
+
+from codalab import settings
+from .models import NewsletterSubscription
+from .forms import NewsletterSubscriptionSignUpForm, NewsletterSubscriptionUnsubscribeForm
+
+
+def newsletter_signup(request):
+    form = NewsletterSubscriptionSignUpForm(request.POST or None)
+
+    if form.is_valid():
+        instance = form.save(commit=False)
+        if NewsletterSubscription.objects.filter(email=instance.email).exists():
+            messages.warning(request, 'This email already signed up for newsletters',
+                             'alert alert-warning alert-dismissible')
+        else:
+            context = {
+                "site": Site.objects.get_current(),
+                "user": request.user,
+            }
+
+            data = {
+                "email_address": instance.email,
+                "status": "subscribed",
+            }
+
+            requests.post(
+                settings.MAILCHIMP_MEMBERS_ENDPOINT_NEWSLETTER,
+                auth=("", settings.MAILCHIMP_API_KEY),
+                data=json.dumps(data)
+            )
+
+            instance.save()
+            messages.success(request, 'You have been added to the Codalab newsletter',
+                             'alert alert-success alert-dismissible')
+            subject = "Thank you for joining the Codalab newsletter"
```

Comment:
Should put this logic into a small helper function to reduce DRY, similar to:
https://github.com/codalab/codalab-competitions/blob/49ce6a472b6ab04511c04636b59db4df8fda9ef0/codalab/apps/api/views/competition_views.py#L203

Reply:
Added a helper function in latest push for `_send_mail()`